src/code/z_skelanime.c

```diff
@@ -1,176 +1,1680 @@
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    Gfx* gfx[4];
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    func_800C6AC4(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 773);
+    Matrix_Push();
+    limbEntry = SEGMENTED_TO_VIRTUAL(limbTable[limbIndex]);
+    limbIndex++;
+    rot = actorDrawTable[limbIndex];
+
+    pos.x = limbEntry->translation.x;
+    pos.y = limbEntry->translation.y;
+    pos.z = limbEntry->translation.z;
+
+    dList = limbEntry->displayLists[dListIndex];
+    if ((updateMtxFunc == NULL) || (updateMtxFunc(globalCtx, limbIndex, &dList, &pos, &rot, actor) == 0)) {
+        Matrix_TranslateThenRotateZYX(&pos, &rot);
+        if (dList != NULL) {
+            do {
+                if (1) {
+                    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_skelanime.c", 0x325),
+                              G_MTX_LOAD);
+
+                    gSPDisplayList(gfxCtx->polyOpa.p++, dList);
+                }
+            } while (0);
+        }
+    }
+
+    if (appendDlistFunc != NULL) {
+        appendDlistFunc(globalCtx, limbIndex, &dList, &rot, actor);
+    }
+
+    if (limbEntry->firstChildIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimb(globalCtx, limbEntry->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                              appendDlistFunc, actor, dListIndex);
+    }
+
+    Matrix_Pull();
+
+    if (limbEntry->nextLimbIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimb(globalCtx, limbEntry->nextLimbIndex, limbTable, actorDrawTable, updateMtxFunc,
+                              appendDlistFunc, actor, dListIndex);
+    }
+
+    func_800C6B54(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 0x335);
+}
+
+void SkelAnime_LodDraw(GlobalContext* globalCtx, u32* limbIndex, Vec3s* actorDrawTable,
+                       SkelAnime_LimbUpdateMatrix updateMtxFunc, SkelAnime_LimbAppendDlist appendDlistFunc,
+                       Actor* actor, s32 dListIndex) {
+    SkelLimbEntry* limbEntry;
+    char pad[4];
+    Gfx* dList;
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    Gfx* gfx[4];
+
+    if (limbIndex == NULL) {
+        osSyncPrintf("\x1b[31m");
+        osSyncPrintf("Si2_Lod_draw():skelがNULLです。\n");
+        osSyncPrintf("\x1b[m");
+        return;
+    }
+    gfxCtx = globalCtx->state.gfxCtx;
+    func_800C6AC4(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 0x351);
+    Matrix_Push();
+
+    limbEntry = SEGMENTED_TO_VIRTUAL(*limbIndex);
+    pos.x = (f32)actorDrawTable->x;
+    pos.y = (f32)actorDrawTable->y;
+    pos.z = (f32)actorDrawTable->z;
+
+    rot = actorDrawTable[1];
+    dList = limbEntry->displayLists[dListIndex];
+
+    if ((updateMtxFunc == NULL) || (updateMtxFunc(globalCtx, 1, &dList, &pos, &rot, actor) == 0)) {
+        Matrix_TranslateThenRotateZYX(&pos, &rot);
+        if (dList != NULL) {
+            gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_skelanime.c", 0x371),
```

Comment:
some line numbers here and there are in hex

Reply:
Also fixed in f2c65d6